utils/sysfs/sysfs.go

```diff
@@ -279,7 +290,23 @@ func (fs *realSysFs) GetNetworkStatValue(dev string, stat string) (uint64, error
 
 func (fs *realSysFs) GetCaches(id int) ([]os.FileInfo, error) {
 	cpuPath := fmt.Sprintf("%s%d/cache", cacheDir, id)
-	return ioutil.ReadDir(cpuPath)
+	dir, err := os.ReadDir(cpuPath)
```

Comment:
I know this is an incompatible change, but it would make sense to convert this to return `[]os.DirEntry` instead.

Otherwise, practically, what happens here, is a `stat(2)` call to every entry, which is kinda expensive. More to say, it seems the only user of this function is in utils/sysinfo ([I checked](https://sourcegraph.com/search?q=context:global+sysfs.GetCaches+-file:%28%5E%7C/%29vendor/+&patternType=standard&sm=1&groupBy=repo)), so while this is a public API, apparently it has no users.

Reply:
There are three other uses of `toFileInfo()` that would have to be fixed. While I understand you concern and I believe that it's valid, I'm hesitant to do this as part of this PR and I would prefer to address it in future.